docs/proposal-speculative-execution.md

```
@@ -0,0 +1,308 @@
+      "on_fail": "replan"
+    }
+  ],
+  "done_when": [{"var": "temp_c", "op": "lt", "val": 25.0}]
+}
+```
+
+Guard language (deliberately minimal in v1): numeric comparisons `gt/lt/in`, discrete equality `eq`, rate-of-change `delta_lt`, time bounds `within_s`, AND composition only (no OR). The limited expressiveness is a feature, not a bug — decidable, evaluable in ~100 lines of C, statically auditable.
+
+### 4.2 Guarded-commit execution semantics
+
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)